javascript/script.js

```diff
@@ -0,0 +1,185 @@
+        );
+        return;
+      } else {
+        return inputNumValue;
+      }
+    }
+  };
+
+  //                       STEP 3 :
+  //  function for confirm bar for whether the user wants lowercase characters in password
+
+  function lowerCaseOption() {
+    const option = confirm(
+      "Press OK if you want your password to have lowercase characters"
+    );
+    return option;
+  }
+
+  //                       STEP 4 :
+  //  function for confirm bar for whether the user wants uppercase characters in password
+
+  function upperCaseOption() {
+    const option = confirm(
+      "Press OK if you want your password to have uppercase characters"
+    );
+    return option;
+  }
+
+  //                       STEP 5 :
+  //  function for confirm bar for whether the user wants numbers characters in password
+
+  function numericOption() {
+    const option = confirm(
+      "Press OK if you want your password to have Numbers"
+    );
+    return option;
+  }
+
+  //                       STEP 6 :
+  //  function for confirm bar for whether the user wants special characters in password
+  function specialCharacterOption() {
+    const option = confirm(
+      "Press OK if you want your password to have special characters"
+    );
+    return option;
+  }
+
+  //                       STEP 7 :
+  //  in a function declare a list of lowercase , uppercase , numbers and special characters in a string
+  // convert the string into an array and then generate a random
+
+  function randomLowerCaseCharacter() {
+    const lowerCaseCharacters = "abcdefghijklmnopqrstuvwxyz";
+    const lowerCaseArray = lowerCaseCharacters.split("");
+    const generateRandomLowerCC =
+      lowerCaseArray[Math.floor(Math.random() * lowerCaseArray.length)];
+    return generateRandomLowerCC;
+  }
+
+  function randomUppercaseCharacter() {
+    const upperCaseCharacters = "abcdefghijklmnopqrstuvwxyz";
+    const upperCaseArray = upperCaseCharacters.toUpperCase().split("");
+    const generateRandomUpperCC =
+      upperCaseArray[Math.floor(Math.random() * upperCaseArray.length)];
+    return generateRandomUpperCC;
+  }
+
+  function randomNumericalValue() {
+    const numericalValue = "0123456789";
+    const numericalArray = numericalValue.split("");
+    const generateRandomNumericalValue =
+      numericalArray[Math.floor(Math.random() * numericalArray.length)];
+    return generateRandomNumericalValue;
+  }
+
+  function randomSpecialCharacter() {
+    const specialCharacters = "@|!#$%&/()=?»«@£§€{}.-;<>_,";
+    const specialCharactersArray = specialCharacters.split("");
+    const generateRandomSpecialCharacter =
+      specialCharactersArray[
+        Math.floor(Math.random() * specialCharactersArray.length)
+      ];
+    return generateRandomSpecialCharacter;
+  }
+
+  //                       STEP 8  :
+  // store the above functions in a variable
+
+  const passwordLength = userPasswordLength();
+
+  const inputLowerCase = lowerCaseOption();
```

Comment:
We need to find a way to only run these functions (the confirm character ones) if the password length question is answered correctly. Currently even if i type an incorrect value in the password length prompt, the programme continues to run, as there is no check here.